Source/Fluids/WarpXFluidContainer.H

```diff
@@ -0,0 +1,172 @@
+     */
+    void centrifugal_source (int lev);
+
+    /**
+     * GatherAndPush introduces the lorentz term in the cold relativistic fluid
+     * equations for a single timestep (dt) using Higuera and Cary Push
+     */
+    /**
+     * \brief Lorentz Momentum Source
+     *
+     * \param[in] lev refinement level
+     * \param[in] Ex Yee electric field (x)
+     * \param[in] Ey Yee electric field (y)
+     * \param[in] Ez Yee electric field (z)
+     * \param[in] Bx Yee magnetic field (x)
+     * \param[in] By Yee magnetic field (y)
+     * \param[in] Bz Yee magnetic field (z)
+     * \param[in] cur_time Current time
+     */
+    void GatherAndPush (int lev,
+        const amrex::MultiFab& Ex, const amrex::MultiFab& Ey, const amrex::MultiFab& Ez,
+        const amrex::MultiFab& Bx, const amrex::MultiFab& By, const amrex::MultiFab& Bz,
+        amrex::Real cur_time);
+
+    /**
+     * DepositCurrent interpolates the fluid current density comps. onto the Yee grid and
+     * sums the contributions to the particle current density
+     */
+    /**
+     * \brief Deposit fluid current density.
+     *
+     * \param[in] lev refinement level
+     * \param[in,out] jx current density MultiFab x comp.
+     * \param[in,out] jy current density MultiFab y comp.
+     * \param[in,out] jz current density MultiFab z comp.
+     */
+    void DepositCurrent (int lev,
+        amrex::MultiFab& jx, amrex::MultiFab& jy, amrex::MultiFab& jz);
+
+    /**
+     * DepositCharge deposits density onto rho
+     */
+    /**
+     * \brief Deposit fluid charge density.
+     *
+     * \param[in] lev refinement level
+     * \param[in,out] rho charge density MultiFab.
+     */
+    void DepositCharge (int lev, amrex::MultiFab &rho, int icomp = 0);
+
+    amrex::Real getCharge () const {return charge;}
+    amrex::Real getMass () const {return mass;}
+
+protected:
+    int species_id;
+    std::string species_name;
+    amrex::Real charge;
+    amrex::Real mass;
+
+    int do_not_push = 0;
+    int do_not_gather = 0;
+    int do_not_deposit = 0;
+    std::string physical_element;
+    PhysicalSpecies physical_species;
+
+    // Parser for external fields
+    std::string m_B_ext_s = "none";
+    std::string m_E_ext_s = "none";
+    // Parser for B_external on the particle
+    std::unique_ptr<amrex::Parser> m_Bx_parser;
+    std::unique_ptr<amrex::Parser> m_By_parser;
+    std::unique_ptr<amrex::Parser> m_Bz_parser;
+    // Parser for E_external on the particle
+    std::unique_ptr<amrex::Parser> m_Ex_parser;
+    std::unique_ptr<amrex::Parser> m_Ey_parser;
+    std::unique_ptr<amrex::Parser> m_Ez_parser;
+    amrex::ParserExecutor<4> m_Exfield_parser;
+    amrex::ParserExecutor<4> m_Eyfield_parser;
+    amrex::ParserExecutor<4> m_Ezfield_parser;
+    amrex::ParserExecutor<4> m_Bxfield_parser;
+    amrex::ParserExecutor<4> m_Byfield_parser;
+    amrex::ParserExecutor<4> m_Bzfield_parser;
+
+    std::unique_ptr<PlasmaInjector> plasma_injector;
+
+public:
```

Comment:
Maybe you could add a one-line comment to explain what these structures are, e.g.
```
// MultiFabs that contain the density (N) and momentum density (NU) of this fluid species, for each refinement level
```

Reply:
Added.